From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
Date: Tue, 29 Jul 2025 15:41:22 +0000
Subject: [PATCH] Revert "[passwords/android] Move & reduce visibility of some
 settings classes"

This reverts commit 3eeac69d90c9c498ac9daadd06c1e7dd83ead273.
---
 chrome/android/chrome_java_sources.gni          |  1 +
 .../settings}/ManualCallbackDelayer.java        |  4 ++--
 chrome/android/junit/BUILD.gn                   |  4 ++++
 .../settings}/DialogManagerTest.java            |  2 +-
 .../settings}/EnsureAsyncPostingRule.java       | 17 +++++------------
 .../SingleThreadBarrierClosureTest.java         |  2 +-
 .../settings}/TimedCallbackDelayerTest.java     |  2 +-
 .../browser/password_manager/android/BUILD.gn   |  6 ++++++
 .../settings}/CallbackDelayer.java              |  4 ++--
 .../settings}/DialogManager.java                |  4 ++--
 .../settings}/ExportErrorDialogFragment.java    |  5 +----
 .../settings}/NonCancelableProgressBar.java     |  5 +----
 .../settings}/SingleThreadBarrierClosure.java   |  4 ++--
 .../settings}/TimedCallbackDelayer.java         |  4 ++--
 .../android/pwm_disabled/BUILD.gn               | 13 -------------
 .../PasswordCsvDownloadFlowController.java      |  3 +++
 16 files changed, 34 insertions(+), 46 deletions(-)
 rename chrome/{browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled => android/java/src/org/chromium/chrome/browser/password_manager/settings}/ManualCallbackDelayer.java (86%)
 rename chrome/{browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled => android/junit/src/org/chromium/chrome/browser/password_manager/settings}/DialogManagerTest.java (98%)
 rename chrome/{browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled => android/junit/src/org/chromium/chrome/browser/password_manager/settings}/EnsureAsyncPostingRule.java (69%)
 rename chrome/{browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled => android/junit/src/org/chromium/chrome/browser/password_manager/settings}/SingleThreadBarrierClosureTest.java (96%)
 rename chrome/{browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled => android/junit/src/org/chromium/chrome/browser/password_manager/settings}/TimedCallbackDelayerTest.java (96%)
 rename chrome/browser/password_manager/android/{pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled => java/src/org/chromium/chrome/browser/password_manager/settings}/CallbackDelayer.java (83%)
 rename chrome/browser/password_manager/android/{pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled => java/src/org/chromium/chrome/browser/password_manager/settings}/DialogManager.java (98%)
 rename chrome/browser/password_manager/android/{pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled => java/src/org/chromium/chrome/browser/password_manager/settings}/ExportErrorDialogFragment.java (95%)
 rename chrome/browser/password_manager/android/{pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled => java/src/org/chromium/chrome/browser/password_manager/settings}/NonCancelableProgressBar.java (90%)
 rename chrome/browser/password_manager/android/{pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled => java/src/org/chromium/chrome/browser/password_manager/settings}/SingleThreadBarrierClosure.java (92%)
 rename chrome/browser/password_manager/android/{pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled => java/src/org/chromium/chrome/browser/password_manager/settings}/TimedCallbackDelayer.java (87%)

diff --git a/chrome/android/chrome_java_sources.gni b/chrome/android/chrome_java_sources.gni
index e2052066abaf6..caae504fc6a65 100644
--- a/chrome/android/chrome_java_sources.gni
+++ b/chrome/android/chrome_java_sources.gni
@@ -866,6 +866,7 @@ chrome_java_sources = [
   "java/src/org/chromium/chrome/browser/password_manager/PasswordManagerDialogViewBinder.java",
   "java/src/org/chromium/chrome/browser/password_manager/PasswordManagerErrorMessageHelperBridge.java",
   "java/src/org/chromium/chrome/browser/password_manager/PasswordManagerLauncher.java",
+  "java/src/org/chromium/chrome/browser/password_manager/settings/ManualCallbackDelayer.java",
   "java/src/org/chromium/chrome/browser/payments/AddressEditor.java",
   "java/src/org/chromium/chrome/browser/payments/AutofillContact.java",
   "java/src/org/chromium/chrome/browser/payments/ChromePaymentRequestFactory.java",
diff --git a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/ManualCallbackDelayer.java b/chrome/android/java/src/org/chromium/chrome/browser/password_manager/settings/ManualCallbackDelayer.java
similarity index 86%
rename from chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/ManualCallbackDelayer.java
rename to chrome/android/java/src/org/chromium/chrome/browser/password_manager/settings/ManualCallbackDelayer.java
index eed24b82f75cc..bbe3d63b21c85 100644
--- a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/ManualCallbackDelayer.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/password_manager/settings/ManualCallbackDelayer.java
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-package org.chromium.chrome.browser.pwm_disabled;
+package org.chromium.chrome.browser.password_manager.settings;
 
 
 import org.chromium.build.annotations.NullMarked;
@@ -14,7 +14,7 @@ import java.util.List;
  * An implementation of {@link CallbackDelayer} for tests. It runs callbacks after a manual signal.
  */
 @NullMarked
-final class ManualCallbackDelayer implements CallbackDelayer {
+public final class ManualCallbackDelayer implements CallbackDelayer {
     /** The callbacks to be run within {@link runCallbacksSynchronously}.*/
     private final List<Runnable> mCallbacks = new ArrayList<>();
 
diff --git a/chrome/android/junit/BUILD.gn b/chrome/android/junit/BUILD.gn
index 5fbd788afd05b..e30bb12b239d8 100644
--- a/chrome/android/junit/BUILD.gn
+++ b/chrome/android/junit/BUILD.gn
@@ -896,6 +896,10 @@ if (is_android) {
     sources = [
       "src/org/chromium/chrome/browser/password_manager/PasswordCheckupLauncherTest.java",
       "src/org/chromium/chrome/browser/password_manager/PasswordManagerErrorMessageHelperBridgeTest.java",
+      "src/org/chromium/chrome/browser/password_manager/settings/DialogManagerTest.java",
+      "src/org/chromium/chrome/browser/password_manager/settings/EnsureAsyncPostingRule.java",
+      "src/org/chromium/chrome/browser/password_manager/settings/SingleThreadBarrierClosureTest.java",
+      "src/org/chromium/chrome/browser/password_manager/settings/TimedCallbackDelayerTest.java",
     ]
 
     deps = [ ":chrome_junit_tests_helper" ]
diff --git a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/DialogManagerTest.java b/chrome/android/junit/src/org/chromium/chrome/browser/password_manager/settings/DialogManagerTest.java
similarity index 98%
rename from chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/DialogManagerTest.java
rename to chrome/android/junit/src/org/chromium/chrome/browser/password_manager/settings/DialogManagerTest.java
index d0d13e5835e6b..c987d3978ffa1 100644
--- a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/DialogManagerTest.java
+++ b/chrome/android/junit/src/org/chromium/chrome/browser/password_manager/settings/DialogManagerTest.java
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-package org.chromium.chrome.browser.pwm_disabled;
+package org.chromium.chrome.browser.password_manager.settings;
 
 import static org.junit.Assert.assertEquals;
 import static org.mockito.Mockito.mock;
diff --git a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/EnsureAsyncPostingRule.java b/chrome/android/junit/src/org/chromium/chrome/browser/password_manager/settings/EnsureAsyncPostingRule.java
similarity index 69%
rename from chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/EnsureAsyncPostingRule.java
rename to chrome/android/junit/src/org/chromium/chrome/browser/password_manager/settings/EnsureAsyncPostingRule.java
index 979cfbfc5491c..e526aa9b290de 100644
--- a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/EnsureAsyncPostingRule.java
+++ b/chrome/android/junit/src/org/chromium/chrome/browser/password_manager/settings/EnsureAsyncPostingRule.java
@@ -2,26 +2,22 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-package org.chromium.chrome.browser.pwm_disabled;
+package org.chromium.chrome.browser.password_manager.settings;
 
 import org.junit.rules.ExternalResource;
 import org.robolectric.Robolectric;
 import org.robolectric.util.Scheduler;
 
-import org.chromium.build.annotations.NullMarked;
-import org.chromium.build.annotations.Nullable;
-
 /**
  * A Rule for ensuring that posted tasks are run asynchronously (and on demand). To do so, the rule
  * pauses the Roboelectric scheduler before running the test and restores its state afterwards.
  */
-@NullMarked
-class EnsureAsyncPostingRule extends ExternalResource {
+public class EnsureAsyncPostingRule extends ExternalResource {
     /** Remembers the paused state of the scheduler so that it can be restored after the test. */
     private boolean mWasSchedulerPaused;
 
-    /** A reference to the scheduler which needs to be paused. */
-    private @Nullable Scheduler mScheduler;
+    /** A reference to the scheduler which needs to be paused.*/
+    private Scheduler mScheduler;
 
     @Override
     protected void before() {
@@ -34,9 +30,6 @@ class EnsureAsyncPostingRule extends ExternalResource {
 
     @Override
     protected void after() {
-        if (!mWasSchedulerPaused) {
-            assert mScheduler != null;
-            mScheduler.unPause();
-        }
+        if (!mWasSchedulerPaused) mScheduler.unPause();
     }
 }
diff --git a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/SingleThreadBarrierClosureTest.java b/chrome/android/junit/src/org/chromium/chrome/browser/password_manager/settings/SingleThreadBarrierClosureTest.java
similarity index 96%
rename from chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/SingleThreadBarrierClosureTest.java
rename to chrome/android/junit/src/org/chromium/chrome/browser/password_manager/settings/SingleThreadBarrierClosureTest.java
index 834f32f543771..affaa5c83363f 100644
--- a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/SingleThreadBarrierClosureTest.java
+++ b/chrome/android/junit/src/org/chromium/chrome/browser/password_manager/settings/SingleThreadBarrierClosureTest.java
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-package org.chromium.chrome.browser.pwm_disabled;
+package org.chromium.chrome.browser.password_manager.settings;
 
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.never;
diff --git a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/TimedCallbackDelayerTest.java b/chrome/android/junit/src/org/chromium/chrome/browser/password_manager/settings/TimedCallbackDelayerTest.java
similarity index 96%
rename from chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/TimedCallbackDelayerTest.java
rename to chrome/android/junit/src/org/chromium/chrome/browser/password_manager/settings/TimedCallbackDelayerTest.java
index 060d7b2f251f2..f94f8944f95ef 100644
--- a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/TimedCallbackDelayerTest.java
+++ b/chrome/android/junit/src/org/chromium/chrome/browser/password_manager/settings/TimedCallbackDelayerTest.java
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-package org.chromium.chrome.browser.pwm_disabled;
+package org.chromium.chrome.browser.password_manager.settings;
 
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.never;
diff --git a/chrome/browser/password_manager/android/BUILD.gn b/chrome/browser/password_manager/android/BUILD.gn
index 26a7693876da3..95f6161679498 100644
--- a/chrome/browser/password_manager/android/BUILD.gn
+++ b/chrome/browser/password_manager/android/BUILD.gn
@@ -220,6 +220,12 @@ android_library("utils_java") {
     "java/src/org/chromium/chrome/browser/password_manager/HelpUrlLauncher.java",
     "java/src/org/chromium/chrome/browser/password_manager/LoginDbDeprecationUtilBridge.java",
     "java/src/org/chromium/chrome/browser/password_manager/PasswordExportLauncher.java",
+    "java/src/org/chromium/chrome/browser/password_manager/settings/CallbackDelayer.java",
+    "java/src/org/chromium/chrome/browser/password_manager/settings/DialogManager.java",
+    "java/src/org/chromium/chrome/browser/password_manager/settings/ExportErrorDialogFragment.java",
+    "java/src/org/chromium/chrome/browser/password_manager/settings/NonCancelableProgressBar.java",
+    "java/src/org/chromium/chrome/browser/password_manager/settings/SingleThreadBarrierClosure.java",
+    "java/src/org/chromium/chrome/browser/password_manager/settings/TimedCallbackDelayer.java",
   ]
 
   srcjar_deps = [ ":utils_jni_headers" ]
diff --git a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/CallbackDelayer.java b/chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/CallbackDelayer.java
similarity index 83%
rename from chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/CallbackDelayer.java
rename to chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/CallbackDelayer.java
index cdf67a92ed781..8619e4e9435cc 100644
--- a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/CallbackDelayer.java
+++ b/chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/CallbackDelayer.java
@@ -2,13 +2,13 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-package org.chromium.chrome.browser.pwm_disabled;
+package org.chromium.chrome.browser.password_manager.settings;
 
 import org.chromium.build.annotations.NullMarked;
 
 /** This is an interface for delaying running of callbacks. */
 @NullMarked
-interface CallbackDelayer {
+public interface CallbackDelayer {
     /**
      * Run a callback after a delay specific to a particular implementation. The callback is always
      * run asynchronously.
diff --git a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/DialogManager.java b/chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/DialogManager.java
similarity index 98%
rename from chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/DialogManager.java
rename to chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/DialogManager.java
index 7b56beb696dc3..d1dd63525fc38 100644
--- a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/DialogManager.java
+++ b/chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/DialogManager.java
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-package org.chromium.chrome.browser.pwm_disabled;
+package org.chromium.chrome.browser.password_manager.settings;
 
 import androidx.annotation.IntDef;
 import androidx.fragment.app.DialogFragment;
@@ -23,7 +23,7 @@ import java.lang.annotation.RetentionPolicy;
  * testing purposes.
  */
 @NullMarked
-final class DialogManager {
+public final class DialogManager {
     /**
      * Contains the reference to a {@link android.app.DialogFragment} between the call to {@link
      * show} and dismissing the dialog.
diff --git a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/ExportErrorDialogFragment.java b/chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/ExportErrorDialogFragment.java
similarity index 95%
rename from chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/ExportErrorDialogFragment.java
rename to chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/ExportErrorDialogFragment.java
index 4081729ad2aca..38099b9c41477 100644
--- a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/ExportErrorDialogFragment.java
+++ b/chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/ExportErrorDialogFragment.java
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-package org.chromium.chrome.browser.pwm_disabled;
+package org.chromium.chrome.browser.password_manager.settings;
 
 import android.app.Dialog;
 import android.content.DialogInterface;
@@ -20,9 +20,6 @@ import org.chromium.chrome.browser.password_manager.R;
 /**
  * Shows the dialog that explains to the user the error which just happened during exporting and
  * optionally helps them to take actions to fix that (learning more, retrying export).
- *
- * <p>PasswordCsvDownloadFlowControllerTest requires the class to be public (and @VisibleForTesting
- * can't be used in top-level classes).
  */
 @NullMarked
 public class ExportErrorDialogFragment extends DialogFragment {
diff --git a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/NonCancelableProgressBar.java b/chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/NonCancelableProgressBar.java
similarity index 90%
rename from chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/NonCancelableProgressBar.java
rename to chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/NonCancelableProgressBar.java
index 99cefe7dab085..1d02d1aecc048 100644
--- a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/NonCancelableProgressBar.java
+++ b/chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/NonCancelableProgressBar.java
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-package org.chromium.chrome.browser.pwm_disabled;
+package org.chromium.chrome.browser.password_manager.settings;
 
 import android.app.Activity;
 import android.app.Dialog;
@@ -20,9 +20,6 @@ import org.chromium.components.browser_ui.widget.MaterialProgressBar;
 /**
  * Shows the dialog that informs the user that some operation is ongoing without indicating the
  * progress.
- *
- * <p>PasswordCsvDownloadFlowControllerTest requires the class to be public (and @VisibleForTesting
- * can't be used in top-level classes).
  */
 @NullMarked
 public class NonCancelableProgressBar extends DialogFragment {
diff --git a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/SingleThreadBarrierClosure.java b/chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/SingleThreadBarrierClosure.java
similarity index 92%
rename from chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/SingleThreadBarrierClosure.java
rename to chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/SingleThreadBarrierClosure.java
index 1aefcc51714d4..98db790084b70 100644
--- a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/SingleThreadBarrierClosure.java
+++ b/chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/SingleThreadBarrierClosure.java
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-package org.chromium.chrome.browser.pwm_disabled;
+package org.chromium.chrome.browser.password_manager.settings;
 
 import org.chromium.build.annotations.NullMarked;
 
@@ -12,7 +12,7 @@ import org.chromium.build.annotations.NullMarked;
  * SingleThreadBarrierClosure is only meant to be used on a single thread and is not thread-safe.
  */
 @NullMarked
-final class SingleThreadBarrierClosure implements Runnable {
+public final class SingleThreadBarrierClosure implements Runnable {
     /** Counts the remaining number of runs. */
     private int mRemainingRuns;
 
diff --git a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/TimedCallbackDelayer.java b/chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/TimedCallbackDelayer.java
similarity index 87%
rename from chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/TimedCallbackDelayer.java
rename to chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/TimedCallbackDelayer.java
index d33c0a1835171..249c4f415e115 100644
--- a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/TimedCallbackDelayer.java
+++ b/chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/TimedCallbackDelayer.java
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-package org.chromium.chrome.browser.pwm_disabled;
+package org.chromium.chrome.browser.password_manager.settings;
 
 import android.os.Handler;
 
@@ -10,7 +10,7 @@ import org.chromium.build.annotations.NullMarked;
 
 /** An implementation of {@link CallbackDelayer} which runs callbacks after a fixed time delay. */
 @NullMarked
-final class TimedCallbackDelayer implements CallbackDelayer {
+public final class TimedCallbackDelayer implements CallbackDelayer {
     /** The {@link Handler} used to delay the callbacks. */
     private final Handler mHandler = new Handler();
 
diff --git a/chrome/browser/password_manager/android/pwm_disabled/BUILD.gn b/chrome/browser/password_manager/android/pwm_disabled/BUILD.gn
index 1aef14ff74698..7700125ac4a13 100644
--- a/chrome/browser/password_manager/android/pwm_disabled/BUILD.gn
+++ b/chrome/browser/password_manager/android/pwm_disabled/BUILD.gn
@@ -9,10 +9,6 @@ android_library("java") {
   resources_package = "org.chromium.chrome.browser.pwm_disabled"
 
   sources = [
-    "java/src/org/chromium/chrome/browser/pwm_disabled/CallbackDelayer.java",
-    "java/src/org/chromium/chrome/browser/pwm_disabled/DialogManager.java",
-    "java/src/org/chromium/chrome/browser/pwm_disabled/ExportErrorDialogFragment.java",
-    "java/src/org/chromium/chrome/browser/pwm_disabled/NonCancelableProgressBar.java",
     "java/src/org/chromium/chrome/browser/pwm_disabled/PasswordCsvDownloadDialogController.java",
     "java/src/org/chromium/chrome/browser/pwm_disabled/PasswordCsvDownloadDialogFragment.java",
     "java/src/org/chromium/chrome/browser/pwm_disabled/PasswordCsvDownloadDialogProperties.java",
@@ -22,8 +18,6 @@ android_library("java") {
     "java/src/org/chromium/chrome/browser/pwm_disabled/PasswordManagerUnavailableDialogCoordinator.java",
     "java/src/org/chromium/chrome/browser/pwm_disabled/PasswordManagerUnavailableDialogMediator.java",
     "java/src/org/chromium/chrome/browser/pwm_disabled/PwmDeprecationDialogsMetricsRecorder.java",
-    "java/src/org/chromium/chrome/browser/pwm_disabled/SingleThreadBarrierClosure.java",
-    "java/src/org/chromium/chrome/browser/pwm_disabled/TimedCallbackDelayer.java",
   ]
 
   deps = [
@@ -36,7 +30,6 @@ android_library("java") {
     "//chrome/browser/ui/android/strings:ui_strings_grd",
     "//components/browser_ui/settings/android:java",
     "//components/browser_ui/theme/android:java_resources",
-    "//components/browser_ui/widget/android:java",
     "//components/device_reauth:device_reauth_java_enums",
     "//components/prefs/android:java",
     "//components/strings:components_strings_grd",
@@ -60,14 +53,9 @@ robolectric_library("junit") {
   testonly = true
 
   sources = [
-    "java/src/org/chromium/chrome/browser/pwm_disabled/DialogManagerTest.java",
-    "java/src/org/chromium/chrome/browser/pwm_disabled/EnsureAsyncPostingRule.java",
-    "java/src/org/chromium/chrome/browser/pwm_disabled/ManualCallbackDelayer.java",
     "java/src/org/chromium/chrome/browser/pwm_disabled/PasswordCsvDownloadDialogTest.java",
     "java/src/org/chromium/chrome/browser/pwm_disabled/PasswordCsvDownloadFlowControllerTest.java",
     "java/src/org/chromium/chrome/browser/pwm_disabled/PasswordManagerUnavailableDialogTest.java",
-    "java/src/org/chromium/chrome/browser/pwm_disabled/SingleThreadBarrierClosureTest.java",
-    "java/src/org/chromium/chrome/browser/pwm_disabled/TimedCallbackDelayerTest.java",
   ]
 
   deps = [
@@ -82,7 +70,6 @@ robolectric_library("junit") {
     "//components/browser_ui/settings/android:java",
     "//components/browser_ui/test/android:test_support_java",
     "//components/device_reauth:device_reauth_java_enums",
-    "//third_party/androidx:androidx_annotation_annotation_java",
     "//third_party/androidx:androidx_appcompat_appcompat_java",
     "//third_party/androidx:androidx_fragment_fragment_java",
     "//third_party/junit",
diff --git a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/PasswordCsvDownloadFlowController.java b/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/PasswordCsvDownloadFlowController.java
index 35e67ef833be8..a3358994f2593 100644
--- a/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/PasswordCsvDownloadFlowController.java
+++ b/chrome/browser/password_manager/android/pwm_disabled/java/src/org/chromium/chrome/browser/pwm_disabled/PasswordCsvDownloadFlowController.java
@@ -21,6 +21,9 @@ import org.chromium.chrome.browser.device_reauth.BiometricStatus;
 import org.chromium.chrome.browser.device_reauth.DeviceAuthSource;
 import org.chromium.chrome.browser.device_reauth.ReauthenticatorBridge;
 import org.chromium.chrome.browser.password_manager.LoginDbDeprecationUtilBridge;
+import org.chromium.chrome.browser.password_manager.settings.DialogManager;
+import org.chromium.chrome.browser.password_manager.settings.ExportErrorDialogFragment;
+import org.chromium.chrome.browser.password_manager.settings.NonCancelableProgressBar;
 import org.chromium.chrome.browser.preferences.Pref;
 import org.chromium.chrome.browser.profiles.Profile;
 import org.chromium.chrome.browser.pwm_disabled.PwmDeprecationDialogsMetricsRecorder.DownloadCsvDialogType;
